contracts/upgradeable_contracts/erc20_to_native/ForeignBridgeErcToNative.sol

```diff
@@ -102,4 +116,100 @@ contract ForeignBridgeErcToNative is BasicForeignBridge, ERC20Bridge, OtherSideB
         emit TokensSwapped(saiContract, erc20token(), curBalance);
         boolStorage[storageAddress] = true;
     }
+
+    function saiTopContract() internal pure returns (ISaiTop) {
+        return ISaiTop(0x9b0ccf7C8994E19F39b2B4CF708e0A7DF65fA8a3);
+    }
+
+    function isTokenSwapAllowed(uint256 _ts) public view returns (bool) {
+        uint256 esTs = saiTopContract().caged();
+        if (esTs > 0 && _ts > esTs) {
+            return false;
+        }
+        return true;
+    }
+
+    function halfDuplexErc20token() public pure returns (ERC20) {
+        return ERC20(0x89d24A6b4CcB1B6fAA2625fE562bDD9a23260359);
+    }
+
+    function setMinHDTokenBalance(uint256 _minBalance) external onlyOwner {
+        uintStorage[MIN_HDTOKEN_BALANCE] = _minBalance;
+    }
+
+    function minHDTokenBalance() public view returns (uint256) {
+        return uintStorage[MIN_HDTOKEN_BALANCE];
+    }
+
+    function isHDTokenBalanceAboveMinBalance() public view returns (bool) {
+        if (tokenBalance(halfDuplexErc20token()) > minHDTokenBalance()) {
+            return true;
+        }
+        return false;
+    }
+
+    function tokenBalance(ERC20 _token) internal view returns (uint256) {
+        return _token.balanceOf(address(this));
+    }
+
+    function migrationContract() internal pure returns (IScdMcdMigration) {
+        return IScdMcdMigration(0xc73e0383F3Aff3215E6f04B0331D58CeCf0Ab849);
+    }
+
+    function swapTokens() public {
```

Comment:
Can this method be used in `migrateToMCD`?

Reply:
Updated `migrateToMCD` to call `swapTokens`